sycl/source/detail/scheduler/commands.cpp

```diff
@@ -1741,23 +1741,23 @@ static void ReverseRangeDimensionsForKernel(NDRDescT &NDR) {
   }
 }
 
-pi_result ExecCGCommand::SetKernelParamsAndLaunch(
-    CGExecKernel *ExecKernel,
-    std::shared_ptr<device_image_impl> DeviceImageImpl, RT::PiKernel Kernel,
-    NDRDescT &NDRDesc, std::vector<RT::PiEvent> &RawEvents, RT::PiEvent &Event,
-    const ProgramManager::KernelArgMask &EliminatedArgMask) {
-  std::vector<ArgDesc> &Args = ExecKernel->MArgs;
-  const detail::plugin &Plugin = MQueue->getPlugin();
-
-  auto setFunc = [this, &Plugin, Kernel, &DeviceImageImpl](
-                     detail::ArgDesc &Arg, size_t NextTrueIndex) {
+static pi_result SetKernelParamsAndLaunch(
+    const QueueImplPtr &Queue, std::vector<ArgDesc> &Args,
+    const std::shared_ptr<device_image_impl> &DeviceImageImpl,
+    RT::PiKernel Kernel, NDRDescT &NDRDesc, std::vector<RT::PiEvent> &RawEvents,
+    const EventImplPtr &EventImpl,
+    const ProgramManager::KernelArgMask &EliminatedArgMask,
+    std::function<void *(Requirement *Req)> getMemAllocationFunc) {
+  const detail::plugin &Plugin = Queue->getPlugin();
+
+  auto setFunc = [&Plugin, Kernel, &DeviceImageImpl, &getMemAllocationFunc,
+                  &Queue](detail::ArgDesc &Arg, size_t NextTrueIndex) {
     switch (Arg.MType) {
     case kernel_param_kind_t::kind_stream:
       break;
     case kernel_param_kind_t::kind_accessor: {
       Requirement *Req = (Requirement *)(Arg.MPtr);
-      AllocaCommandBase *AllocaCmd = getAllocaForReq(Req);
-      RT::PiMem MemArg = (RT::PiMem)AllocaCmd->getMemAllocation();
```

Comment:
Could you please add an assert checking that if we are here then `getMemAllocationFunc` is not `nullptr`?